tasks_test.go

```diff
@@ -0,0 +1,180 @@
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+// TestDO represents entry with payload.
+type TestDO struct {
+	Timestamp time.Time      `db:"ts"`
+	Id        uuid.UUID      `db:"id"`
+	Payload   types.JSONText `db:"payload"`
+}
+
+var testObjects = []TestDO{
+	{
+		Id:        uuid.New(),
+		Timestamp: time.Now(),
+		Payload:   types.JSONText("{\"test\": \"json\"}"),
+	},
+	{
+		Id:        uuid.New(),
+		Timestamp: time.Now().Add(time.Hour),
+		Payload:   types.JSONText("{\"test\": \"json2\"}"),
+	},
+}
+
+func triggerIngestionTask(d *Client, dataSourceName string, entries []TestDO) (string, error) {
+	csvEntriesBuff := &bytes.Buffer{}
+
+	err := gocsv.MarshalWithoutHeaders(entries, csvEntriesBuff)
+	if err != nil {
+		return "", err
+	}
+
+	var spec = NewTaskIngestionSpec(
+		SetTaskType("index_parallel"),
+		SetTaskDataSource(dataSourceName),
+		SetTaskTuningConfig("index_parallel", 25000, 5000000),
+		SetTaskIOConfigType("index_parallel"),
+		SetTaskInputFormat("csv", "false", []string{"ts", "id", "payload"}),
+		SetTaskInlineInputData(csvEntriesBuff.String()),
+	)
+	taskID, err := d.Tasks().SubmitTask(spec)
+	return taskID, err
+}
+
+func awaitTaskCompletion(client *Client, taskID string) error {
+	for range time.Tick(100 * time.Millisecond) {
+		res, err := client.Tasks().GetStatus(taskID)
+		if err != nil {
+			return err
+		}
+
+		if res.Status.Status == "RUNNING" {
+			continue
+		}
+		break
+	}
+	return nil
+}
+
+func awaitTaskRunning(client *Client, taskID string) error {
+	for range time.Tick(100 * time.Millisecond) {
+		res, err := client.Tasks().GetStatus(taskID)
+		if err != nil {
+			return err
+		}
+
+		if res.Status.Status == "RUNNING" {
+			return nil
+		}
+	}
+	return errors.New("task has not started")
+}
+
+func runInlineIngestionTask(client *Client, dataSourceName string, entries []TestDO, recordsCount int) error {
+	taskID, err := triggerIngestionTask(client, dataSourceName, entries)
+	if err != nil {
+		return err
+	}
+
+	err = awaitTaskCompletion(client, taskID)
+	if err != nil {
+		return err
+	}
+
+	err = client.Metadata().AwaitDataSourceAvailable(dataSourceName)
+	if err != nil {
+		return err
+	}
+
+	err = client.Metadata().AwaitRecordsCount(dataSourceName, recordsCount)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func TestTaskService(t *testing.T) {
+	// Set up druid containers using docker-compose.
+	compose, err := tc.NewDockerCompose("testdata/docker-compose.yaml")
+	require.NoError(t, err, "NewDockerComposeAPI()")
+
+	// Set up cleanup for druid containers.
+	t.Cleanup(func() {
+		require.NoError(t, compose.Down(context.Background(), tc.RemoveOrphans(true), tc.RemoveVolumes(true), tc.RemoveImagesLocal), "compose.Down()")
+	})
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	// Set up druid service and client.
+	var druidOpts []ClientOption
+	d, err := NewClient("http://localhost:8888", druidOpts...)
```

Comment:
options are empty. Is there any reason why to declare and pass them?

Reply:
fixed.